git-bot/__main__.py

```diff
@@ -21,17 +22,22 @@ async def pr_opened_event(event, gh, *args, **kwargs):
     owner = event.data["repository"]["owner"]["login"]
     repo = event.data["repository"]["name"]
     comments_url = event.data["pull_request"]["comments_url"]
-    
+    import pickle
+    with open('filename.pickle', 'wb') as handle:
+        pickle.dump(event, handle, protocol=pickle.HIGHEST_PROTOCOL)
     # Fetch pull request changes using GitHub API
     changes_url = f"/repos/{owner}/{repo}/pulls/{pr_number}/files"
-    changes_response = await gh.get(changes_url)
+    changes_response = await gh.getitem(changes_url)
     
     # Calculate the total number of additions, deletions, and changes
     total_additions = 0
     total_deletions = 0
     total_changes = 0
 
-    for file in changes_response:
+    for i, file in enumerate(changes_response):
+        import json
+        os.makedirs("changes", exist_ok=True)
+        json.dump(file, open(f"changes/{i}.json", "w"))
```

Comment:
Here’s a suggested change for Insecure direct object reference in file writing:

```suggestion
with open(f"changes/{i}.json", "w") as json_file:
    json.dump(file, json_file)
```